File: competitive-programming/codeforces/round_731_div_3/a.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

#define ll long long
#define output_t void

using point_t = pair<int, int>;

bool ordered(int a, int b, int c)
{
    return (a < b && b < c) || (a > b && b > c);
}

bool same_row(point_t a, point_t b, point_t c)
{
    return (a.second == b.second) and (b.second == c.second);
}

bool same_col(point_t a, point_t b, point_t c)
{
    return (a.first == b.first) and (b.first == c.first);
}

output_t solve(int tc)
{
    point_t a, b, f;

    cin >> a.first >> a.second;
    cin >> b.first >> b.second;
    cin >> f.first >> f.second;

    size_t ans = abs(a.first - b.first) + abs(a.second - b.second);

    bool row_test = same_row(a, b, f) && ordered(a.first, f.first, b.first);
    bool col_test = same_col(a, b, f) && ordered(a.second, f.second, b.second);

    ans = ans + 2 * (row_test || col_test);

    cout << ans << "\n";
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(NULL);

    int tc;

    // tc = 1;
    cin >> tc;

    while (tc--) {
        solve(tc);
    }

    return 0;
}

File: competitive-programming/codeforces/round_731_div_3/a.rs
use std::io::*;

struct Point {
    x: i32,
    y: i32,
}

fn order(a: i32, b: i32, c: i32) -> bool {
    (a < b && b < c) || (a > b && b > c)
}

fn same_row(a: &Point, b: &Point, c: &Point) -> bool {
    (a.y == b.y) && (b.y == c.y)
}

fn same_col(a: &Point, b: &Point, c: &Point) -> bool {
    (a.x == b.x) && (b.x == c.x)
}

fn solve(input: &mut Input) -> () {
    let a = Point {
        x: input.next(),
        y: input.next(),
    };
    let b = Point {
        x: input.next(),
        y: input.next(),
    };
    let f = Point {
        x: input.next(),
        y: input.next(),
    };

    let mut ans: i32 = (a.x - b.x).abs() + (a.y - b.y).abs();

    let row_test: bool = same_row(&a, &b, &f) && order(a.x, f.x, b.x);
    let col_test: bool = same_col(&a, &b, &f) && order(a.y, f.y, b.y);

    if row_test || col_test {
        ans += 2
    };

    println!("{}", ans);
}

fn main() {
    let input = &mut Input::new();

    let tc: i32 = input.next();

    for _ in 1..=tc {
        solve(input);
    }
}

// boilerplate

struct Input {
    buf: Vec<String>,
}

impl Input {
    fn new() -> Self {
        Input { buf: Vec::new() }
    }

    fn next_line(&mut self) -> String {
        let mut line = String::new();
        stdin().read_line(&mut line).expect("ERROR: read_line");
        line = line.trim().to_string();

        if line.len() == 0 {
            return self.next_line();
        }

        line
    }

    fn next<T: std::str::FromStr>(&mut self) -> T {
        if self.buf.is_empty() {
            self.buf = self
                .next_line()
                .split_whitespace()
                .rev() // because of pop()
                .map(String::from)
                .collect();
        }

        self.buf.pop().unwrap().parse().ok().expect("ERROR: parse")
    }
}
